src/main/java/com/assignment/students/controller/ClassController.java

```diff
@@ -19,48 +16,34 @@ public ClassController(ClassService classService) {
         this.classService = classService;
     }
 
-    @PostMapping(path="class")
-    public void addClass(@RequestBody Class subject){
-        try {
-            classService.addClass(subject);
-        }catch (DuplicateKeyException e) {
-            throw new ResponseStatusException(
-                    HttpStatus.BAD_REQUEST, "Class With this name already exists", e);
-        }
+    @PostMapping(path = "/api/v1/classes", consumes = MediaType.APPLICATION_JSON_VALUE)
+    public Class addClass(@RequestBody Class subject) {
+        return classService.addClass(subject);
     }
 
-    @PostMapping(path="class/batch")
-    public void addClass(@RequestBody List<Class> subjectList){
-        try {
-            classService.addClasses(subjectList);
-        }catch (DuplicateKeyException e) {
-            throw new ResponseStatusException(
-                    HttpStatus.BAD_REQUEST, "Class With this name already exists", e);
-        }
+    @PostMapping(path = "/api/v1/classes", consumes = "application/vnd.classes.batch+json")
+    public List<Class> addClass(@RequestBody List<Class> subjectList) {
+        return classService.addClasses(subjectList);
     }
 
-    @GetMapping(path="class")
-    public List<Class> getAllClasses(){
+    @GetMapping(path = "/api/v1/classes")
+    public List<Class> getAllClasses() {
         return classService.getAllClasses();
     }
 
-    @GetMapping(path = "class/{id}")
-    public Class getClassById(@PathVariable long id){
+    @GetMapping(path = "/api/v1/classes/{id}")
+    public Class getClassById(@PathVariable long id) {
         return classService.getClassById(id);
     }
 
-    @PutMapping(path = "class")
-    public void updateClass(@RequestBody Class subject){
-        classService.updateClass(subject);
+    @PutMapping(path = "/api/v1/classes")
```

Comment:
the put mapping should be like the get mapping, end with the {id}. /api/v1/classes/{id}